### DISCUSSION TITLE
`imap <c-j> <Action>(EditorDown)` doesn't work when completions window present.

### DISCUSSION THREAD

**Discussion by sunn4room** [JetBrains/ideavim#896] [Q&A] [answered]
```
sethandler <C-J> a:vim
sethandler <C-K> a:vim
imap <c-j> <Action>(EditorDown)
imap <c-k> <Action>(EditorUp)
" imap <c-j> <Down> 
" imap <c-k> <Up>
```
`<c-j>` is OK, until the completions window present.

> I use the `XWin` preset keymaps, and the origin `<c-j>` map to `insert live template`. After removing the origin map, the problem remains.

I guess that the `ideavim` lost focus when the completions window appeared, and some maps defined in ideavim are overrided.

If I am right, is there a hacker way to handle the problem? 

> My idea version is 2024.1.2
> My ideavim version is 2.12.0

**Comment by OptimusCrime** [comment]
Not sure if I understand 100%, but could you try this instead?:

```
sethandler <C-J> a:ide
sethandler <C-K> a:ide
imap <c-j> <Down> 
imap <c-k> <Up>
```

**Reply by sunn4room** [reply]
I just prefer `<c-h/j/k/l>` to arrow key. I expect that `<c-j/k>` can select next/previous item in completions window. But your config doesn't work.

Somebody has this config.

```
sethandler <C-N> a:vim
sethandler <C-P> a:vim
imap <c-n> <Down>
imap <c-p> <Up>
```

It's OK. But I don't know why the `<c-j>` doesn't work.

**Reply by OptimusCrime** [reply]
I don't know either. Which editor are you using? I could try it myself. I have clearned that the handler needs to be IDE for this to work in IntelliJ, WebStorm and Goland at least.

**Reply by sunn4room** [reply]
OS is `arch linux`
Editor is `intellij idea 2024.1.2 with JRE`

> Not sure if I understand 100%

`<c-k>` can select up around completion items, but `<c-j>` cannot select down in my computer.

The same problem appeared in my friend computer. I don't know where is going wrong.

Thank you very much for your response. And I hope this discussion catches the attention of the testers. If nobody encounters the same problem for days, I will close this discussion. Thanks.

**Reply by sunn4room** [reply]
> I have clearned that the handler needs to be IDE

I noticed already, and problem remains

**Reply by citizenmatt** [reply, marked as answer]
Check out [`'lookupkeys'`](https://github.com/JetBrains/ideavim/wiki/set-commands). The default value includes `<C-J>`, which means it will be processed by the IDE. This probably explains why the Vim mapping for Ctrl+J isn't doing anything.

Also check out the IDE key map settings for the "Up" and "Down" actions in the Settings dialog. This might be useful for experimenting with.
